lms/djangoapps/bulk_email/tasks.py

```diff
@@ -387,25 +393,40 @@ def _filter_optouts_from_recipients(to_list, course_id):
     return to_list, num_optout
 
 
-def _get_source_address(course_id, course_title):
+def _get_source_address(user_pk, course_id, course_title):
     """
     Calculates an email address to be used as the 'from-address' for sent emails.
 
-    Makes a unique from name and address for each course, e.g.
+    Makes a unique from name and address for each course,
 
         "COURSE_TITLE" Course Staff <coursenum-no-reply@courseupdates.edx.org>
 
+    The string is localized based on the user's language preference.
     """
     course_title_no_quotes = re.sub(r'"', '', course_title)
 
     # For the email address, get the course.  Then make sure that it can be used
     # in an email address, by substituting a '_' anywhere a non-(ascii, period, or dash)
     # character appears.
-    from_addr = u'"{0}" Course Staff <{1}-{2}>'.format(
-        course_title_no_quotes,
-        re.sub(r"[^\w.-]", '_', course_id.course),
-        settings.BULK_EMAIL_DEFAULT_FROM_EMAIL
+
+    try:
+        user = User.objects.get(pk=user_pk)
```

Comment:
Cool. Omar, would you mind looking into this, and the testing? I will
follow up next week.

On Wed, Sep 23, 2015 at 2:21 PM, Adam <notifications@github.com> wrote:

> In lms/djangoapps/bulk_email/tasks.py
> <https://github.com/edx/edx-platform/pull/9808#discussion_r40238443>:
>
> >      """
> >      course_title_no_quotes = re.sub(r'"', '', course_title)
> >
> >      # For the email address, get the course.  Then make sure that it can be used
> >      # in an email address, by substituting a '_' anywhere a non-(ascii, period, or dash)
> >      # character appears.
> > -    from_addr = u'"{0}" Course Staff <{1}-{2}>'.format(
> > -        course_title_no_quotes,
> > -        re.sub(r"[^\w.-]", '_', course_id.course),
> > -        settings.BULK_EMAIL_DEFAULT_FROM_EMAIL
> > +
> > +    try:
> > +        user = User.objects.get(pk=user_pk)
>
> @sarina <https://github.com/sarina> , I imaging we do a join when
> grabbing the User objects? I think select_related
> <https://docs.djangoproject.com/en/dev/ref/models/querysets/#django.db.models.query.QuerySet.select_related>
> can be used for that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/9808/files#r40238443>.
>